src/components/modal/Modal.jsx

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import './Modal.css';
+
+const Modal = ({
+  isOpen,
+  onClose,
+  confirmClose,
+  children
+}) => {
+  if (!isOpen) return null;
+
+  const handleClose = () => {
+    if (confirmClose) {
+      confirmClose().then((confirm) => {
+        if (confirm) onClose();
+      });
+    } else {
+      onClose();
+    }
+  };
```

Comment:
I like the use of a promise for the `confirmClose`. However, there are two possible improvements. 

1. Make sure that it works if I pass a confirm close synchronous function (for example if I you do `<Modal confirmClose={() => true} >` it's going to break the modal.
2. Remove the `if` condition.

Do you know how?